ckan/logic/schema.py

```diff
@@ -443,6 +443,22 @@ def user_edit_form_schema(
     return schema
 
 
+@validator_args
+def user_perform_reset_form_schema(
+        not_empty, unicode_safe, user_both_passwords_entered,
+        user_password_validator, user_passwords_match):
+    # (canada fork only): adds schema for PerformResetView
+    # TODO: upstream contrib??
+    schema = default_user_schema()
+
+    schema['password1'] = [text_type, user_both_passwords_entered,
+                           user_password_validator, user_passwords_match]
+    schema['password2'] = [text_type]
```

Comment:
I thought text_type (str) isn't an acceptable validator, or maybe just in 2.10? unicode_only or unicode_safe are reasonable choices.

Reply:
@wardi yeah `text_type` is apparently still used in 2.9. And it is just imported from `six`. This will break in 2.10, but Core uses `unicode_safe` there so won't be an issue. Looks like the `text_type` is only used by some of the user schemas in core code for 2.9, so possibly just and overlooked thing there: https://github.com/ckan/ckan/blob/2.9/ckan/logic/schema.py#L422

Will use `unicode_safe` in upstream contrib and in our 2.10 fork/branch.